File: python_solution/problem6.py
# -*- coding: utf-8 -*-
"""
Spyder Editor

This is a temporary script file.
"""
class Problem6:
    def zigzagConversion(self, s: str, n: int)-> str:
        length = len(s);
        arr = [""] * n
        row = 0;
        down = True;
        for i in range(length):
            arr[row] += (s[i])
    
            if row == n-1:
                down = False
    
            elif row == 0:
                down = True
    
            if down:
                row +=1
    
            else:
                row -=1
    
        result = "".join(arr)
        print(result)
        return result

def main():
    # Read input from the console
    s = input("Enter a string: ")
    n= int(input("Number of Rows: "))

    # Create an instance of Solution and find the length of the longest substring
    solution_prob6 = Problem6()
    result = solution_prob6.zigzagConversion(s,n)

    # Print the result
    print(f"ZigZag conversion of Given String {s} : {result}")

if __name__ == "__main__":
    main()

File: C#Solutions/problem6.cs
using System;

namespace AllProblems
{
    public static class Problem6
    {
        public static void Solve()
        {
            Console.WriteLine("Solution to Problem 6");
            Console.WriteLine("Enter a string to find the ZigZag string:");
            string s = Console.ReadLine();
            Console.WriteLine("Enter a Int number of rows:");
            int numRows = int.Parse(Console.ReadLine());
            string result = zigzagString(s,numRows);
            Console.WriteLine("Longest Palindromic Substring: " + result);
        }

        

        public static string zigzagString(string s, int numRows)
        {
            if (string.IsNullOrEmpty(s))
            {
                return "";
            }

            if(numRows == 1){
            return s;
            
        }
        int len = s.Length;
        string []arr = new string[numRows];
        int row = 0;
        bool down = true;
        for(int i = 0; i < len; i++){
            arr[row] += (s[i]);
    
            if (row == numRows-1){
                down = false;
            }
            else if(row == 0){
                down = true;
            }
            if(down){
                row++;
            }
            else{
                row--;
            }

        }
        string result = string.Join("", arr);
        return result;

        }
    }
}
